dkg/nodesigs.go

```diff
@@ -50,32 +62,64 @@ func newNodeSigBcast(operatorAmt int, bcastComp *bcast.Component) nodeSigBcast {
 }
 
 // broadcastCallback is the default bcast.Callback for nodeSigBcast.
-func (n *nodeSigBcast) broadcastCallback(_ context.Context, _ peer.ID, _ string, msg proto.Message) error {
+func (n *nodeSigBcast) broadcastCallback(ctx context.Context, peer peer.ID, _ string, msg proto.Message) error {
 	response, ok := msg.(*dkgpb.MsgNodeSig)
 	if !ok {
 		return errors.New("invalid node sig type")
 	}
 
-	n.otherSigs <- response
+	sig := response.GetSignature()
+	msgPeerIdx := int(response.GetPeerIndex())
+
+	if (msgPeerIdx == n.nodeIdx.PeerIdx) || (msgPeerIdx < 0 || msgPeerIdx > n.operatorAmt) {
+		return errors.New("wrong peer index", z.Str("peer", peer.String()))
```

Comment:
nit: we just use `invalid` in most cases. Nit, you can also remove the "peer" since that caller of this function knows what peer it is, so no need to tell it. It actually already logs the peer. So only return fields in errors that are unknown to the caller.

Also, just FYI, we have `p2p.PeerName` when logging peer names